uvtools/dspec.py

```diff
@@ -60,11 +60,356 @@ def calc_width(filter_size, real_delta, nsamples):
+                   #The core code of this method will always assume 2d data
+                   if ndim_data == 1:
+                       data = np.asarray([data])
+                       wgts = np.asarray([wgts])
+                   if not filter2d and filter_dim == 0:
+                       data = data.T
+                       wgts = wgts.T
+                   if mode[0] == 'dayenu':
+                       if filter2d:
+                           filter_dim_d = [0, 1]
+                       else:
+                           filter_dim_d = [1]
+                       residual, info = dayenu_filter(x=x, data=data, wgts=wgts, filter_dimensions=filter_dim_d,
+                                                     filter_centers=filter_centers, filter_half_widths=filter_half_widths,
+                                                     filter_factors=suppression_factors, cache=cache)
+                       model = data - residual
+                       if len(mode) > 1:
+                           model, _, info_deconv = fit_basis_2d(x=x, data=model, filter_centers=filter_centers, filter_dims=filter_dim_d,
+                                                                 skip_wgt=skip_wgt, basis=mode[1], method=mode[2], wgts=wgts, basis_options=fitting_options,
+                                                                 filter_half_widths=filter_half_widths, suppression_factors=suppression_factors,
+                                                                 cache=cache, max_contiguous_edge_flags=max_contiguous_edge_flags)
+                           info['info_deconv']=info_deconv
+
+                   elif mode[0] == 'dft' or mode[0] == 'dpss':
+                        if filter2d:
+                            filter_dim_d = [0, 1]
+                        else:
+                            filter_dim_d = [1]
+                        model, residual, info = fit_basis_2d(x=x, data=data, filter_centers=filter_centers, filter_dims=filter_dim_d,
+                                                            skip_wgt=skip_wgt, basis=mode[0], method=mode[1], wgts=wgts, basis_options=fitting_options,
+                                                            filter_half_widths=filter_half_widths, suppression_factors=suppression_factors,
+                                                            cache=cache, max_contiguous_edge_flags=max_contiguous_edge_flags)
+                   elif mode[0] == 'clean':
+                        #Unpack all of the clean parameters from
+                        #fitting_options. This is to preserve default behavior
+                        #in high_pass_fourier_filter and if I had my way,
+                        #I'd get rid of them.
+                        #However, keeping them preserves
+                        #high_pass_fourier_filter default behavior
+                        #for those who were raised on it.
+                        if not 'tol' in fitting_options:
```

Comment:
I think an easier way to do this is to define a module level dictionary at the top of this file
```
CLEAN_DEFAULTS = dict(tol=1e-9, window='none', alpha=0, ...)
```
and then in this function, do one of the two, not sure which is better
```
for k in CLEAN_DEFAULTS:
  if k in fitting_options:
    locals().update(k, fitting_options[k])
  else:
    locals().update(k, CLEAN_DEFAULTS[k])
```
and then the attrs are defined in the local environment
or you can do something like
```
for k in CLEAN_DEFAULTS:
  if k not in fitting_options:
    fitting_options[k] = CLEAN_DEFAULTS[k]
```
and then just call these in this block as `fitting_options['tol']` rather than unpack them all.

Problem with both of these is that there is no typo protection, in otherwords, if I fed `fitting_options = {'tole': 1e-2}`, then the code __would not fail__, it would not see `tol` and replace it with the default and run per usual, leading to silent failure. 

The best way to resolve this, in my opinion, is to keep a separate `CLEAN` function (perhaps `high_pass_fourier_filter`?) that does have default params in its signature, and in this case you just do something like this
```
high_pass_fourier_filter(*args, **fitting_options)
```
we can discuss this next week if you like

Reply:
Should we discuss this on the analysis telecon?

Reply:
I think I can protect against typo failure by checking the keys. What do you think of this idea?